contracts/preset/ERC721PsiMysteryBox.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+/**
+  ______ _____   _____ ______ ___  __ _  _  _ 
+ |  ____|  __ \ / ____|____  |__ \/_ | || || |
+ | |__  | |__) | |        / /   ) || | \| |/ |
+ |  __| |  _  /| |       / /   / / | |\_   _/ 
+ | |____| | \ \| |____  / /   / /_ | |  | |   
+ |______|_|  \_\\_____|/_/   |____||_|  |_|   
+                                              
+                                            
+ */
+
+pragma solidity ^0.8.0;
+
+import "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
+import "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+import "fpe-map/contracts/FPEMap.sol";
+
+import "../ERC721Psi.sol";
+
+abstract contract ERC721PsiMysteryBox is VRFConsumerBaseV2, ERC721Psi {
+    using Strings for uint256;
+
+    uint32 private constant _callbackGasLimit = 200000;
+
+    uint256 private seed;
+
+    event RandomnessRequest(uint256 requestId);
+
+
+    constructor() {
+        VRFConsumerBaseV2(_coordinator());
+    }
+
+    /**
+     * @dev The metadata URI before revealing.
+     */
+    function _unrevealedURI() internal pure virtual returns (string memory) {
+        return "";
+    }
+
+    /**
+     * @dev See {IERC721Metadata-tokenURI}.
+     */
+    function tokenURI(uint256 tokenId) public view virtual override returns (string memory) {
+        require(_exists(tokenId), "ERC721Psi: URI query for nonexistent token");
+
+        if(seed == 0) {
+            return _unrevealedURI();
+        }
+
+        string memory baseURI = _baseURI();
+
+        uint256 metadataId = FPEMap.fpeMappingFeistelAuto(tokenId, seed, _maxSupply()); 
+        
+        return bytes(baseURI).length > 0 ? string(abi.encodePacked(baseURI, 
+            metadataId.toString()
```

Comment:
Have you considered appending file extension here? For instance ".json" is mandatory if using IPFS for storage.

Maybe something like this:
```suggestion
            metadataId.toString(), _extension()
```